From c589487cf566fd7d55a87fd89b923711466858b6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ra=C3=BAl=20Cumplido?= <raulcumplido@gmail.com>
Date: Mon, 14 Apr 2025 15:04:44 +0200
Subject: [PATCH 7/8] GH-46111: [C++][CI] Fix boost 1.88 on MinGW (#46113)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

### Rationale for this change

Boost has remove their `include/boost/process/v1.hpp`

### What changes are included in this PR?

Instead of using their old include file (which was a wrapper for the different includes) add the individual includes.

### Are these changes tested?

Via CI.

### Are there any user-facing changes?

No
* GitHub Issue: #46111

Authored-by: Raúl Cumplido <raulcumplido@gmail.com>
Signed-off-by: Raúl Cumplido <raulcumplido@gmail.com>
---
 cpp/src/arrow/testing/process.cc | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/cpp/src/arrow/testing/process.cc b/cpp/src/arrow/testing/process.cc
index 57df0196c1..1cf15bcb31 100644
--- a/cpp/src/arrow/testing/process.cc
+++ b/cpp/src/arrow/testing/process.cc
@@ -68,7 +68,24 @@
 #      define BOOST_USE_WINDOWS_H = 1
 #    endif
 #    ifdef BOOST_PROCESS_HAVE_V1
-#      include <boost/process/v1.hpp>
+#      include <boost/process/v1/args.hpp>
+#      include <boost/process/v1/async.hpp>
+#      include <boost/process/v1/async_system.hpp>
+#      include <boost/process/v1/child.hpp>
+#      include <boost/process/v1/cmd.hpp>
+#      include <boost/process/v1/env.hpp>
+#      include <boost/process/v1/environment.hpp>
+#      include <boost/process/v1/error.hpp>
+#      include <boost/process/v1/exe.hpp>
+#      include <boost/process/v1/group.hpp>
+#      include <boost/process/v1/handles.hpp>
+#      include <boost/process/v1/io.hpp>
+#      include <boost/process/v1/pipe.hpp>
+#      include <boost/process/v1/search_path.hpp>
+#      include <boost/process/v1/shell.hpp>
+#      include <boost/process/v1/spawn.hpp>
+#      include <boost/process/v1/start_dir.hpp>
+#      include <boost/process/v1/system.hpp>
 #    else
 #      include <boost/process.hpp>
 #    endif
